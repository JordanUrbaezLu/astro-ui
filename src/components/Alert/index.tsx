/* istanbul ignore file */
export * from "./Alert";
