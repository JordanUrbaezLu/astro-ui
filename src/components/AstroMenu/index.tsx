export * from "./AstroMenu";
export * from "./AstroMenuItem";
