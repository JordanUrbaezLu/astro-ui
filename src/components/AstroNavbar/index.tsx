export * from "./AstroNavbar";
export * from "./AstroNavbarItem";
