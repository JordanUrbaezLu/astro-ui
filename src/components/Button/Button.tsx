import * as React from "react";
import styles from "./Button.module.scss";
import classNames from "classnames";
import { Typography } from "../Typography/Typography";
import { Ripple } from "../Ripple";

export type ButtonVariant = "primary" | "secondary";
export type ButtonSize = "small" | "medium" | "large";
export type ButtonType = "button" | "submit" | "reset";

export interface ButtonProps
  extends React.ComponentPropsWithoutRef<"button"> {
  /**
   * The content for the Button
   */
  children: string;
  /**
   * If the Button is disabled
   *
   * @default false
   */
  disabled?: boolean;
  /**
   * The leading icon for the Button
   */
  leadingIcon?: React.ReactNode;
  /**
   * The callback fired when the Button is clicked
   */
  onClick?: () => void;
  /**
   * The size of the Button
   *
   * @default "medium"
   */
  size?: ButtonSize;
  /**
   * The trailing icon for the Button
   */
  trailingIcon?: React.ReactNode;
  /**
   * The type of the Button
   */
  type?: ButtonType;
  /**
   * The variant of the Button
   *
   * @default "primary"
   */
  variant?: ButtonVariant;
}

/**
 * @public
 *
 * @description
 *
 * The Button component is used to allows user to take an action.
 */
export const Button = React.forwardRef<
  HTMLButtonElement,
  ButtonProps
>((props, ref) => {
  const {
    className,
    children,
    disabled = false,
    leadingIcon,
    onClick,
    size = "medium",
    trailingIcon,
    type,
    variant = "primary",
    ...rest
  } = props;

  return (
    <button
      className={classNames(
        className,
        styles.button,
        disabled && styles.disabled,
        size === "small"
          ? styles.small
          : size === "large"
          ? styles.large
          : styles.medium,
        variant === "secondary" ? styles.secondary : styles.primary
      )}
      disabled={disabled}
      onClick={onClick}
      ref={ref}
      type={
        type === "reset"
          ? "reset"
          : type === "submit"
          ? "submit"
          : "button"
      }
      {...rest}
    >
      {!disabled && <Ripple position="cursor" />}
      <Typography
        className={styles.typography}
        type={
          size === "large" ? "h1" : size === "medium" ? "h2" : "h3"
        }
        variant={variant}
      >
        {leadingIcon && (
          <span className={styles.leadingIcon}>{leadingIcon}</span>
        )}
        {children}
        {trailingIcon && (
          <span className={styles.trailingIcon}>{trailingIcon}</span>
        )}
      </Typography>
    </button>
  );
});

Button.displayName = "Button";
