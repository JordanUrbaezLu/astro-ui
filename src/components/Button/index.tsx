/* istanbul ignore file */
export * from "./Button";
