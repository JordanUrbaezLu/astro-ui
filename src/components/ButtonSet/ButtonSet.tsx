import * as React from "react";
import classNames from "classnames";
import styles from "./ButtonSet.module.scss";

export interface ButtonSetProps extends React.ComponentPropsWithoutRef<"div"> {
  /**
   * The buttons in the Button Set
   */
  children: React.ReactNode;
}

export const ButtonSet: React.FC<ButtonSetProps> = (props) => {
  const { className, children, ...rest } = props;

  return (
    <div className={classNames(styles.buttonSet, className)} {...rest}>
      {children}
    </div>
  );
};
