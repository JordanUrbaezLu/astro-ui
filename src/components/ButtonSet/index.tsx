/* istanbul ignore file */
export * from "./ButtonSet";
