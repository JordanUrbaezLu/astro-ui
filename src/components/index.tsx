export * from "./Alert";
export * from "./Button";
export * from "./Checkbox";
export * from "./DatePicker";
export * from "./Link";
export * from "./Menu";
export * from "./Tooltip";
export * from "./Select";
export * from "./Switch";
export * from "./TextField";
