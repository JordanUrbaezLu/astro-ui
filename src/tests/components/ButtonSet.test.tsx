import { render, screen } from "@testing-library/react";
import { Button } from "../../components/Button";
import { ButtonSet } from "../../components/ButtonSet";
import { axe } from "jest-axe";

describe("ButtonSet", () => {
  test("Should render component correctly.", () => {
    const { container } = render(
      <ButtonSet>
        <Button>Button</Button>
        <Button>Button</Button>
        <Button>Button</Button>
      </ButtonSet>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe("Accessibility", () => {
  test("Should have no accessibility violations.", async () => {
    const { container } = render(
      <ButtonSet>
        <Button>Button</Button>
        <Button>Button</Button>
        <Button>Button</Button>
      </ButtonSet>
    );
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});

describe("Props", () => {
  test("Should render children correctly.", () => {
    render(
      <ButtonSet>
        <Button>Button</Button>
        <div>Div</div>
      </ButtonSet>
    );
    expect(screen.getByRole("list")).not.toHaveTextContent("Div");
    expect(screen.getByRole("list")).toHaveTextContent("Button");
  });

  test("Should render flex direction correctly.", () => {
    render(
      <ButtonSet flexDirection="column">
        <Button>Button</Button>
      </ButtonSet>
    );
    expect(screen.getByRole("list")).toHaveClass("buttonSetColumn");
  });
});
