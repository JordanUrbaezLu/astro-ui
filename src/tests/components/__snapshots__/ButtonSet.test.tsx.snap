// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonSet Should render component correctly. 1`] = `
<ul
  class="buttonSet"
  role="list"
>
  <li
    class="buttonSetItem"
    role="listitem"
  >
    <button
      class="button medium primary"
      type="button"
    >
      <div
        class="container"
        role="presentation"
      />
      <div
        class="typography typography typographyh2 primary normal"
      >
        Button
      </div>
    </button>
  </li>
  <li
    class="buttonSetItem"
    role="listitem"
  >
    <button
      class="button medium primary"
      type="button"
    >
      <div
        class="container"
        role="presentation"
      />
      <div
        class="typography typography typographyh2 primary normal"
      >
        Button
      </div>
    </button>
  </li>
  <li
    class="buttonSetItem"
    role="listitem"
  >
    <button
      class="button medium primary"
      type="button"
    >
      <div
        class="container"
        role="presentation"
      />
      <div
        class="typography typography typographyh2 primary normal"
      >
        Button
      </div>
    </button>
  </li>
</ul>
`;
